### PR TITLE
Add support for MutinyVerticle

### PR DESCRIPTION
**Pull request by cescoffier**
Just add support for `MutinyVerticle` - a type of verticle (Vert.x components) using the Mutiny API.

This PR contains the feature, the documentation, and tests.

### CODE REVIEW

**Comment by cescoffier** [comment]
The CI issue is related, I will fix it later Today.

**Comment by cescoffier** [comment]
The issue should be fixed.

**Comment by cescoffier** [comment]
Rebased to see if the CI is happier now...

File: docs/src/main/asciidoc/vertx.adoc [hunk 1]
```diff
@@ -498,6 +498,86 @@ Then, create the native executable with:
...
+import javax.inject.Inject;
+
+@ApplicationScoped
+public class VerticleDeployer {
+
+    @Inject Vertx vertx;
+    @Inject MyBeanVerticle verticle;
+
+    public void init(@Observes StartupEvent e) {
+         vertx.deployVerticle(verticle).await().indefinitely();
```
Comment (mkouba) [inline docs/src/main/asciidoc/vertx.adoc]: It should quite easy to deploy those verticles automatically. 

Something like:

```java
void deploy(@Observes StartupEvent event, Vertx vertx, Instance<AbstractVerticle> verticles) {
   for (AbstractVerticle verticle : verticles) {
     vertx.deployVerticle(verticle).await().indefinitely();
   }
}
```

File: docs/src/main/asciidoc/vertx.adoc [hunk 2]
```diff
@@ -498,6 +498,86 @@ Then, create the native executable with:
...
+    @Override
+    public Uni<Void> asyncStart() {
+        return vertx.eventBus().consumer(address)
+                .handler(m -> m.replyAndForget("hello"))
+                .completionHandler();
+    }
+}
+====
+
+Don't inject the `vertx` instance in an injected verticle.
```
Comment (mkouba) [inline docs/src/main/asciidoc/vertx.adoc]: I'd reword this sentence. Maybe something like _"You don't have to inject the `Vertx` instance but instead leverage the instance stored in the protected field of `AbstractVerticle`"_...

File: docs/src/main/asciidoc/vertx.adoc (hunk 1 above)
Comment (cescoffier) [reply docs/src/main/asciidoc/vertx.adoc]: ah ah! I didn't know this! Let me update the documentation.

File: docs/src/main/asciidoc/vertx.adoc (hunk 2 above)
Comment (cescoffier) [reply docs/src/main/asciidoc/vertx.adoc]: Yes, that's better.

File: docs/src/main/asciidoc/vertx.adoc (hunk 1 above)
Comment (cescoffier) [reply docs/src/main/asciidoc/vertx.adoc]: Updated the doc and the test! 
Soon I will be a CDI master... _soon_

**Comment by cescoffier** [comment]
Yet another rebase to see if the CI is happy now...

**Comment by cescoffier** [comment]
@mkouba Should be ok now.

**Review by mkouba — APPROVED** [review]

**Review by gsmet — APPROVED** [review]
